packages/icon-build-helpers/src/builders/react/next/typescript.js

```diff
@@ -46,9 +46,8 @@ async function writeModuleTypes(modules, outDir) {
       templates.banner +
       '\n' +
       "import type { CarbonIconType } from './CarbonIcon';\n" +
-      'export const ' +
-      m.name +
-      ': CarbonIconType;\n';
+      `const ${m.name}: CarbonIconType;\n` +
+      `export default ${m.name}\n`;
```

Comment:
FYI, I see this locally when copying/pasting this change to my local environment:

![Screenshot 2023-10-26 at 8 30 35 AM](https://github.com/carbon-design-system/carbon/assets/10718366/d4eb37e9-9597-4dcf-8630-eba44030be8a)

Reply:
@metonym thanks for spotting! I added the missing declare keyword and it seems to be better now. Please confirm if this also helps in your setup.

Reply:
LGTM